src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetCommandHandler.cs

```diff
@@ -100,6 +102,23 @@ public CommandState GetCommandState(TabKeyCommandArgs args)
             return CommandState.Available;
         }
 
+        public CommandState GetCommandState(AutomaticLineEnderCommandArgs args, Func<CommandState> nextCommandHandler)
+        {
+            return nextCommandHandler();
+        }
+
+        public void ExecuteCommand(AutomaticLineEnderCommandArgs args, Action nextCommandHandler, CommandExecutionContext executionContext)
+        {
+            AssertIsForeground();
+            if (AreSnippetsEnabled(args)
+                && args.TextView.Properties.TryGetProperty(typeof(AbstractSnippetExpansionClient), out AbstractSnippetExpansionClient snippetExpansionClient))
```

Comment:
nit: consider extracting helper for this as tihs pattern appears a lot.  NBD if you don't

Reply:
➡️ Leaving this for now. My new code is only a minority of cases where the pattern appears so opting to adhere to existing pattern for simplicity while I focus on the other parts.